gvm/protocols/gmpv7.py

```diff
@@ -2215,34 +2225,433 @@ def modify_role(self, role_id, comment=None, name=None, users=None):
 
         return self._send_xml_command(cmd)
 
-    def modify_scanner(self, scanner_id, host, port, scanner_type, **kwargs):
-        cmd = self._generator.modify_scanner_command(scanner_id, host, port,
-                                                     scanner_type, kwargs)
-        return self.send_command(cmd)
+    def modify_scanner(self, scanner_id, host, port, scanner_type,
+                       comment=None, name=None, ca_pub=None,
+                       credential_id=None):
+        """Modifies an existing scanner.
 
-    def modify_schedule(self, schedule_id, **kwargs):
-        cmd = self._generator.modify_schedule_command(schedule_id, kwargs)
-        return self.send_command(cmd)
+        Arguments:
+            scanner_id (str): UUID of scanner to modify.
+            host (str): Host of the scanner.
+            port (str): Port of the scanner.
+            scanner_type (str): Type of the scanner.
+                '1' for OSP, '2' for OpenVAS (classic) Scanner.
+            comment (str, optional): Comment on scanner.
+            name (str, optional): Name of scanner.
+            ca_pub (str, optional): Certificate of CA to verify scanner's
+                certificate.
+            credential_id (str, optional): UUID of the client certificate credential
+                for the Scanner.
+        """
+        if not scanner_id:
+            raise RequiredArgument('modify_scanner requires a scanner_id argument')
+        if not host:
+            raise RequiredArgument('modify_scanner requires a host argument')
+        if not port:
+            raise RequiredArgument('modify_scanner requires a port argument')
+        if not scanner_type:
+            raise RequiredArgument('modify_scanner requires a type argument')
+
+        cmd = XmlCommand('modify_scanner')
+        cmd.set_attribute('scanner_id', scanner_id)
+        cmd.add_element('host', host)
+        cmd.add_element('port', port)
+        cmd.add_element('type', scanner_type)
```

Comment:
Could you add a check if the scanner_type is '1' or '2' and raise an InvalidArgument error if not?